src/dispatcher/index.js

```diff
@@ -0,0 +1,175 @@
+const resultQueue = process.env.RESULTSQUEUE;
+const rabbitHost = process.env.RABBITHOST;
+const rabbitUser = process.env.RABBITUSER;
+const rabbitPass = process.env.RABBITPASS;
+
+let channel;
+
+/**
+ * Listen for a job message and trigger the different involved workers.
+ * The entries in the job array are handled sequential, starting from top. The `type` of
+ * a worker configuration is used as the queue name to push the job and its configuration to.
+ * In the following example, the first worker will be called by passing a message to the
+ * queue named `download`, while the message itself is the original job message extend by
+ * a `nextJob` entry which indicates which task shall be handled next.
+ * If tasks finish succesfully, they usually return values which will then be appended to
+ * the specific task configuration via the `outputs` array.
+ *
+ * @example job message content:
+ * 
+  {
+    "job": [
+      {
+        "id": 123,
+        "type": "download-new-data-from-url",
+        "inputs": [
+          "https://deb.debian.org/debian/dists/bullseye/main/installer-amd64/current/images/cdrom/debian-cd_info.tar.gz",
+          "/home/data/"
+        ]
+      },
+      {
+        "id": 456,
+        "type": "gunzip",
+        "inputs": [
+          {
+            "outputOfId": 123,
+            "outputIndex": 0
+          }
+        ]
+      }
+    ]
+  }
+ */
+const dispatcher = async() => {
+  const connection = await amqp.connect({
+    hostname: rabbitHost,
+    username: rabbitUser,
+    password: rabbitPass,
+    heartbeat: 60
+  }).catch(errorAndExit);
+
+  channel = await connection.createChannel().catch(errorAndExit);
+
+  // TODO: implement deadLetterExchange
+  channel.assertQueue(workerQueue, {
+    durable: true
+  });
+  channel.assertQueue(resultQueue, {
+    durable: true
+  });
+
+  log(`Dispatcher waiting for messages in ` +
+    `${workerQueue} and ${resultQueue}.`);
+
+  channel.consume(
+    workerQueue,
+    handleNextTask,
+    {
+      noAck: false
+    }
+  );
+
+  channel.consume(
+    resultQueue,
+    handleResults,
+    {
+      noAck: false
+    }
+  );
+};
+
+/**
+ * Determine the next task from job list and sends it to worker queue.
+ * If there is no task left, finish the execution with success.
+ * @param {Object} msg The message
+ */
+const handleNextTask = (msg) => {
+  let nextTaskEntry;
+  let firstIteration = false;
+
+  try {
+    const job = JSON.parse(msg.content.toString());
+    const chain = job.job;
+    log('Received a job configuration, invoking workers...')
+
+    // validate
+    if (!chain || chain.length < 1) {
+      errorAndExit(msg, 'Invalid arguments given' + job);
+    }
+
+    // create unique ID if necessary (on first run)
+    if (!job.id) {
+      job.id = parseInt(new Date() * Math.random(), 10);
+      firstIteration = true;
+    }
+
+    // find next job entry that has not run yet (no status set)
+    nextTaskEntry = chain.find(task => !task.status);
```

Comment:
we should define what we understand from `status`. If it boolean, it should rather be called like `completed` or so. Otherwise it might become confusing

Reply:
changed to explicit string content check